### PR TITLE
Add options to enable tracing for ZippyDB Laser etc.

### PR DESCRIPTION
**Pull request by nmjnmjnmj**
This is part of  https://www.internalfb.com/diff/D24276354

Enable tracing for zippydb and other services(Laser, etc.) which use rocksdb.

In the past couple months, we've canaried this to zippydb.disagg/zippydb.disagg.wildcard/zippydb.shadow.wildcard.flash etc. tiers and successfully collected multiple traces.

We'd like to checkin this to to collect traces in a more scalable way by just Canary the config changes.
It will enable us to roll out the config change and enable tracing in a larger scale.

### CODE REVIEW

**Review by ajkr — CHANGES_REQUESTED** [review]
Can we update the title/description to describe only what changed in RocksDB? Internal service names, tier names, and tool links are confusing to have in the public commit history.

**Review by zhichao-cao — CHANGES_REQUESTED** [review]
I'm not comfortable to put them in db_options. They are not related to the db behavior. Also, we have query tracing, IO tracing, and block cache tracing. Those options are miss leading. We have the options specifically for tracing: struct TraceOptions. uint64_t max_trace_file_size is already there. And I believe, some of the options you introduced, can be the member of TraceOptions. Another issue is, enable_trace and trace_is_on is not the option. We can introduce the function in tracer to get the status.

File: include/rocksdb/options.h [hunk 1]
```diff
@@ -375,6 +375,20 @@ struct DBOptions {
...
+  std::string trace_path = "";
+
+  // This specifies the traces file name.
+  std::string trace_name_postfix = "";
+
+  // Max trace file size in byte
+  uint64_t max_trace_file_size = 1 * 1024 * 1024 * 1024;
+
+  // If true, the database will be created if it is missing.
+  bool enable_trace = false;
```
Comment (zhichao-cao) [inline include/rocksdb/options.h:388]: Why the database will be created if it is missing?

File: include/rocksdb/options.h [hunk 2]
```diff
@@ -375,6 +375,20 @@ struct DBOptions {
...
   DBOptions* IncreaseParallelism(int total_threads = 16);
 #endif  // ROCKSDB_LITE
+  // This specifies the traces file path.
+  std::string trace_path = "";
+
+  // This specifies the traces file name.
+  std::string trace_name_postfix = "";
+
+  // Max trace file size in byte
+  uint64_t max_trace_file_size = 1 * 1024 * 1024 * 1024;
```
Comment (riversand963) [inline include/rocksdb/options.h:385]: I suggest that you do `1ULL * 1024 * 1024 * 1024` or `static_cast<uint64_t>(1) * 1024 * 1024 * 10224`.
Should people in the future change the default to something like `4 * 1024 * 1024 * 1024`, this value will overflow.

**Comment by riversand963** [comment]
@nmjnmjnmj do you still plan to work on this?

**Comment by nmjnmjnmj** [comment]
@zhichao-cao no, moved whole thing to zippydb side, will abandon this one

**Pull request closed without merging**